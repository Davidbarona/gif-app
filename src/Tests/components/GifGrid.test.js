import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../Hooks/useFetchGifs'

// realice a fake called into the useFetchgifs
jest.mock('../../Hooks/useFetchGifs')


describe("Test on GifGrid Component",()=>{
    
    const category = 'one punch'
    
    
    test('should render GifGrid Correctly', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true,
        })
        
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
     })

     test('should show itmes when it loading images useFetchGifs', () => { 
        
        const gifs = [{
            id:'ABC',
            url:'https://www.localhost.anything.jpg',
            title:'testing'
        },
        {
            id:'AB',
            url:'https://www.localhost.anything.jpg',
            title:'testing'
        }]
        
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false,
        })

        const wrapper = shallow(<GifGrid category={category} />)
        // create a snapshot about GifGrid 
        expect(wrapper).toMatchSnapshot()
        // if exists some one p tag mmust be false 
        expect(wrapper.find('p').exists()).toBe(false)
        // if there is 1 or more elements then compare with the lenght of the array 
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
       
     })
})