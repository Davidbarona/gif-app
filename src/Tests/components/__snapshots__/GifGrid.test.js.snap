// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGrid Component should render GifGrid Correctly 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__swing"
  >
    one punch
  </h3>
  <p
    className="card animate__animated animate__backInLeft"
  >
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test on GifGrid Component should show itmes when it loading images useFetchGifs 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__swing"
  >
    one punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="testing"
      url="https://www.localhost.anything.jpg"
    />
    <GifGridItem
      id="AB"
      key="AB"
      title="testing"
      url="https://www.localhost.anything.jpg"
    />
  </div>
</Fragment>
`;
